pinot-broker/src/main/java/org/apache/pinot/broker/api/AccessControl.java

```diff
@@ -42,21 +46,58 @@ default boolean hasAccess(RequesterIdentity requesterIdentity) {
 
   /**
    * Fine-grained access control on parsed broker request. May check table, column, permissions, etc.
-   *
+   * The default implementation is kept to have backward compatibility with the existing implementations
    * @param requesterIdentity requester identity
    * @param brokerRequest broker request (incl query)
    *
    * @return {@code true} if authorized, {@code false} otherwise
    */
-  boolean hasAccess(RequesterIdentity requesterIdentity, BrokerRequest brokerRequest);
+
+  default boolean hasAccess(RequesterIdentity requesterIdentity, BrokerRequest brokerRequest) {
+    return true;
+  }
+
+  /**
+   * Verify access control on parsed broker request. May check table, column, permissions, etc.
+   * The default implementation returns a {@link BasicAuthorizationResultImpl} with the result of the hasAccess() of
+   * the implementation
+   *
+   * @param requesterIdentity requester identity
+   * @param brokerRequest broker request (incl query)
+   *
+   * @return {@code AuthorizationResult} with the result of the access control check
+   */
+  default AuthorizationResult verifyAccess(RequesterIdentity requesterIdentity, BrokerRequest brokerRequest) {
```

Comment:
(minor) Since the result is called `AuthorizationResult`, consider renaming the method to be `authorize()`?